File: LeetCode/6. Zigzag Conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1 or numRows >= len(s):
            return s
        rows = [''] * numRows
        cycle_len = 2 * (numRows-1)

        for i,char in enumerate(s):
            row = min(i%cycle_len, cycle_len - (i%cycle_len))
            rows[row] += char
        return ''.join(rows)

File: LeetCode/6. Zigzag Conversion.js
/**
 * @param {string} s
 * @param {number} numRows
 * @return {string}
 */
var convert = function(s, numRows) {
    if(numRows===1 || s.length<=numRows) return s;

    let currentRow=0;
    let rows = new Array(numRows).fill("");
    let goingDown= false;

    for(let c of s){
        rows[currentRow]+= c;

        if(currentRow===0 || currentRow === numRows-1){
            goingDown = ! goingDown;
        }
        currentRow += goingDown? 1: -1;
    }
    return rows.join('');
};
